ui/app/AppLayouts/Chat/stores/MessageStore.qml

```diff
@@ -11,13 +11,49 @@ QtObject {
     readonly property bool loadingHistoryMessagesInProgress: root.chatSectionModule? root.chatSectionModule.loadingHistoryMessagesInProgress : false
     readonly property int newMessagesCount: messagesModel ? messagesModel.newMessagesCount : 0
     readonly property bool messageSearchOngoing: messageModule ? messageModule.messageSearchOngoing : false
+    readonly property bool amIChatAdmin: d.amIChatAdmin
+    readonly property bool isPinAllowedForAllMembers: d.isPinAllowedForAllMembers
+    readonly property string chatId: d.chatId
+    readonly property int chatType: d.chatType
+    readonly property string chatColor: d.chatColor
+    readonly property string chatIcon: d.chatIcon
 
     onMessageModuleChanged: {
         if(!messageModule)
             return
         root.messagesModel = messageModule.model
     }
 
+    readonly property QtObject d: QtObject {
+        //These properties are used to avoid calling the nim functions too often
+        property bool amIChatAdmin: messageModule ? messageModule.amIChatAdmin() : false
+        property bool isPinAllowedForAllMembers: messageModule ? messageModule.pinMessageAllowedForMembers() : false
+        property string chatId: messageModule ? messageModule.getChatId() : ""
+        property int chatType: messageModule ? messageModule.getChatType() : 0
+        property string chatColor: messageModule ? messageModule.getChatColor() : ""
+        property string chatIcon: messageModule ? messageModule.getChatIcon() : ""
+
+        function refreshData() {
+            if(messageModule) {
+                amIChatAdmin = messageModule.amIChatAdmin()
+                isPinAllowedForAllMembers = messageModule.pinMessageAllowedForMembers()
+                chatId = messageModule.getChatId()
+                chatType = messageModule.getChatType()
+                chatColor = messageModule.getChatColor()
+                chatIcon = messageModule.getChatIcon()
+            }
+        }
+
+        property Timer refreshTimer: Timer {
+            interval: 60000 //1 min
+            repeat: true
+            running: true
+            onTriggered: {
+                d.refreshData()
```

Comment:
Uhm... not sure I like this ever running, hardcoded timer :/

Reply:
Isn't there a better solution?

Reply:
Fair enough, please create a followup task for this

Reply:
But does that mean that if we change chat, those properties might not get updated for 1 minute?
If that's so, I could be considered and admin on a channel I,m not an admin and create some very big confusion (probably no actually big issues because the backend has protections, but still)

Reply:
It's true, but still might be better than initially. Since the backend doesn't have proper signals for these properties the user could be considered admin until the delegate is destroyed and recreated (could be much more than 1 minute).

This should be a temporary solution..The proper way to do it is to refactor the backend to catch the chat settings updates and send signals to the MessageStore.

Reply:
It definitely is a hackish thing to do, I agree. Will drop this change from this PR and prepare another one with proper signals.

Reply:
Added the signals is this PR #9075